src/wait-for.js

```diff
@@ -94,6 +94,10 @@ function waitFor(
 
       if (!usingFakeTimers) {
         clearInterval(intervalId)
+        observer.disconnect()
+
+        // Call also the disconnect method on setImmediate for projects not using the MutationObersver implementation of jsdom.
+        // But using for example `@sheerun/mutationobserver-shim` which  provokes infinite loop when being called immediately on `onDone` function
```

Comment:
This looks fine to me. I'm only concerned about the implications of calling disconnect twice. I guess it's fine, but I'm not sure it's worthwhile having code that is only around to support a shim that nobody should be using (because they should be using a modern version of jsdom or a real browser environment). Anyone want to argue against just removing the next line?

Reply:
I have commented and on the issue and proposing two solutions:
- remove the line as you suggest
- check if we are using MutationObserver of `jsdom` or another custom impl
https://github.com/testing-library/dom-testing-library/issues/800#issuecomment-720379391
https://github.com/testing-library/dom-testing-library/issues/800#issuecomment-720437358

Reply:
If nobody should use shim, it's good for me to remove it.
Thank you Kent and Marco, I'm gonna remove it asap 

Does someone know what I should do for the breaking CI ? (if there is something I can do)